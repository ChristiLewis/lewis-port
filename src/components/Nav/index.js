//THIS NAV RETURNS A SINGLE JSX ELEMENT----> THE HEADER

// import React from 'react';
import React, { useEffect } from "react";
// import React, { useEffect } from 'react';
//REACT REJECTED - NEED A HELPER
import { capitalizeFirstLetter } from "../../utils/helpers";

function Nav(props) {
    const {
        categories = [],
        setCurrentCategory,
        currentCategory,
    } = props;

    useEffect(() => {
        document.title = capitalizeFirstLetter(currentCategory.name);
    }, [currentCategory]);

    return (
        <header className="flex-row px-1">
            <h2>
                <a data-testid="link" href="/">
                    <span role="img" aria-label="classical_building, computer">
                        üèõÔ∏è||üíª
                    </span>
                    Phygital!
                </a>
            </h2>
            <nav>
                <ul className="flex-row">
                    <li className="mx-2">
                        <a data-testid="about" href="#about">
                            About me
                        </a>
                    </li>
                    <li className="mx-2">
                        <span>Contact</span>
                    </li>
                    {categories.map((category) => (
                        <li className={`mx-1 ${currentCategory.name === category.name
                            }`} key={category.name}>
                            <span onClick={() => {
                                setCurrentCategory(category)
                                // setContactSelected(false);
                            }}
                            >
                                {capitalizeFirstLetter(category.name)}
                            </span>
                        </li>
                    ))}
                </ul>
            </nav>
        </header>
    );
}

export default Nav;