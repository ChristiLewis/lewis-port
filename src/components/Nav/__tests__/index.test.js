//THIS FILE GOES INTO EACH SRC/COMPONENTS/COMPONENTNAME-PASCALCASE/__TESTS__/INDEX.TEST.JS
import React from 'react';

//IMPORT THE REACT TESTING LIBRARY
import { render, cleanup } from '@testing-library/react';

//IMPORT THE JEST-DOM PACKAGE
import '@testing-library/jest-dom/extend-expect';

//IMPORT THE COMPONENT
import Nav from '..';

//TEST TO HANDLE PROPS FOR THE NAV HERE UNDER THE IMPORT STATEMENTS
const categories = [
    {
        name: "business-finance",
        description:
            "Digital asset tracker app; physical offices, a bank, and a medical project"
    }

]
const mockCurrentCategory = jest.fn();
const mockSetCurrentCategory = jest.fn();

//CONFIGURE THE TEST ENV BY CALLING CLEANUP() VIA AFTEREACH GLOBAL FUNCTION FROM JEST
afterEach(cleanup);

//CREATE A TEST

describe('Nav component', () => {
    // BASELINE Test
    it('renders', () => {
        render(<Nav
            categories={categories}
            setCurrentCategory={mockSetCurrentCategory}
            currentCategory={mockCurrentCategory}
        />);
    });

    //CREATE A TEST CASE TO COMPARE SNAPSHOTS OF THE DOM NODE STRUCTURE- HERE INSIDE THE DESCRIBE CALLBACK FUNCTION BODY- BENEATH THE RENDER TEST
    it('matches snapshot', () => {
        //RETURN THE SNAPSHOT
        const { asFragment } = render(<Nav
            categories={categories}
            setCurrentCategory={mockSetCurrentCategory}
            currentCategory={mockCurrentCategory}
        />);
        //COMPARE - CONTRAST W/ GOAL TO MATCH
        expect(asFragment()).toMatchSnapshot();
    });

})

//CREATE EMOJI VISIBILITY TEST VIA A NEW DESCRIBE FUNCTION
describe('emoji is visible', () => {
    //CHECKS IF EMOJI IS INSERTED INTO THE H2 ELEMENT IN THE NAV COMPONENT
    it('inserts emoji into the h2', () => {
        //QUERY TO RETURN THE EMOJI VIA ITS SPAN ELEMENT
        const { getByLabelText } = render(<Nav
            categories={categories}
            setCurrentCategory={mockSetCurrentCategory}
            currentCategory={mockCurrentCategory}
        />);
        //AND ACCESSIBILITY ARIA LABEL VIA A CUSTOM MATCHER
        expect(getByLabelText('classical_building, computer')).toHaveTextContent('🏛️||💻');
    });
})

//CREATE THE VISIBILITY TEST
describe('links are visible', () => {
    it('inserts text into the links', () => {
        //ARRANGE THE QUERIES
        const { getByTestId } = render(<Nav
            categories={categories}
            setCurrentCategory={mockSetCurrentCategory}
            currentCategory={mockCurrentCategory}
        />);
        //ADD ASSERTION VIA ARIA LABEL VIA A CUSTOM MATCHER
        expect(getByTestId('link')).toHaveTextContent('Phygital!');
        expect(getByTestId('about')).toHaveTextContent('About me');
    });
})













