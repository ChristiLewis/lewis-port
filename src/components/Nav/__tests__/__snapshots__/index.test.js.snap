// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="flex-row px-1"
  >
    <h2>
      <a
        data-testid="link"
        href="/"
      >
        <span
          aria-label="classical_building, computer"
          role="img"
        >
          🏛️||💻
        </span>
        Phygital!
      </a>
    </h2>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-2"
        >
          <a
            data-testid="about"
            href="#about"
          >
            About me
          </a>
        </li>
        <li
          class="mx-2"
        >
          <span>
            Contact
          </span>
        </li>
        <li
          class="mx-1 false"
        >
          <span>
            Business-finance
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
